content/integrate/redis-data-integration/data-pipelines/supported-types.md

```diff
@@ -0,0 +1,525 @@
+---
+Title: Supported data types by source
+alwaysopen: false
+categories:
+- docs
+- integrate
+- rs
+- rdi
+description: Learn about supported data types for each source database.
+group: di
+linkTitle: Supported data types
+summary: Redis Data Integration keeps Redis in sync with the primary database in near
+  real time.
+type: integration
+weight: 80
+---
+
+The sections below describe the data types supported by RDI for each source database.
+There are also some cross‑cutting considerations that apply to all source databases.
+
+## Oracle
+
+### Core Supported Types
+
+RDI (via the Debezium Oracle connector) supports Oracle’s core scalar types:
+
+- **Numeric**: `NUMBER`, `DECIMAL`
+- **Character**: `CHAR`, `VARCHAR2`, `NCHAR`, `NVARCHAR2`
+- **Temporal**: `DATE`, `TIMESTAMP`, `TIMESTAMP WITH TIME ZONE`, `TIMESTAMP WITH LOCAL TIME ZONE`
+
+**Behavior & mapping:**
+
+- `NUMBER(p,s)` / `DECIMAL(p,s)`  
+  - Ingested with full precision and scale.  
+  - **Redis Hash**: stored as a string, e.g. `4521398.56` → `"4521398.56"`.  
+  - **RedisJSON**: stored as a numeric JSON value by default, or as a string if configured.
+
+- `CHAR` / `VARCHAR2` / `NCHAR` / `NVARCHAR2`  
+  - Stored as plain strings in Redis (UTF‑8 preserved).
+
+- `DATE`, `TIMESTAMP`  
+  - Converted to epoch **milliseconds**.  
+  - **Redis Hash**: stored as a string, e.g. `"1529507596945"`.  
+  - **RedisJSON**: stored as a number.  
+
+- `TIMESTAMP WITH TIME ZONE`  
+  - Normalized to UTC, then converted to epoch milliseconds.
+
+### Large Objects (LOBs)
+
+Oracle LOBs are supported **with additional configuration**:
+
+- **Types**: `CLOB`, `NCLOB`, `BLOB` (and LOB‑backed `XMLTYPE`).
+- By default, Debezium does **not** send full LOB contents in change events.
+
+**Required config (RDI / Debezium side):**
+
+```properties
+# In the Oracle source connector / RDI advanced properties
+lob.enabled=true
+```
+
+- When `lob.enabled=true`:
+  - New values for `CLOB` / `NCLOB` / `BLOB` / `XMLTYPE` are sent in events.
+  - Unchanged LOB values in update events may be represented by placeholders (connector optimization).
+
+**Mapping:**
+
+- `CLOB` / `NCLOB` → Unicode string (entire text) in Redis.
+- `BLOB` → Base64‑encoded string (by default).  
+- In Redis Hashes, these are string fields; in RedisJSON they are JSON strings.
+
+### Binary & RAW
+
+- **Types**: `BLOB`, `RAW`, `LONG RAW` (see unsupported below).
+
+By default, Debezium uses a **binary handling mode** (commonly `base64`):
+
+```properties
+binary.handling.mode=base64   # typical default
+```
+
+- Oracle `RAW` / `BLOB` values:
+  - Emitted as Base64 strings (e.g. `"hello"` bytes → `"aGVsbG8="`).
+  - Stored as those encoded strings in Redis Hash fields or RedisJSON string properties.
+
+### XMLTYPE
+
+Oracle `XMLTYPE` is backed by LOBs and **needs extra setup**:
```

Comment:
We need to verify with R&D if we have this jars already included in RDI 
CC: @ZdravkoDonev-redis